addons/medical_engine/functions/fnc_getHitpointArmor.sqf

```diff
@@ -0,0 +1,68 @@
+#include "script_component.hpp"
+/*
+ * Author: Pterolatypus
+ * Checks a unit's equipment to calculate the total armor on a hitpoint
+ *
+ * Arguments:
+ * 0: Unit <OBJECT>
+ * 1: Hitpoint <STRING>
+ * 2: Disable Caching <BOOLEAN> (Default: false)
+ *
+ * Return Value:
+ * Total armor for the given hitpoint <ARRAY>
+ * 0: Armor <NUMBER>
+ * 1: ExplosionShielding <NUMBER>
+ *
+ * Example:
+ * [player, "HitChest"] call ace_medical_engine_fnc_getHitpointArmor
+ *
+ * Public: No
+ */
+
+params ["_unit","_hitpoint"];
+
+// Structural damage is inferred differently, not needed here
+if (_hitpoint in ["", "#structural"]) exitwith {[1,1]};
+
+// Get base values from the unit class
+private _unitCfg = configFile >> "CfgVehicles" >> (typeOf _unit);
+private _armor = getNumber (_unitCfg >> "armor");
+private _explosion = getNumber (_unitCfg >> "explosionShielding");
```

Comment:
Is this actually how it works though? I am pretty sure that the uniformClass determines the armor and explosive (and armorStructural) values and overwrites the unit completely...

Reply:
You are correct, uniform armor does override base class (tested by modding a CSAT uniform up to 1000 armor and putting it on a blufor unit).
New version correctly uses `nakedUniform` for this instead.